Minor changes to #547 (directly to connectStorimod branch)

NightMadness opened the pull request:
**Reverted:** _1. Switch string to Optional<String>_
2. Removed pokemonUniqueID from IVScanResult
3. Added log to PokemonShareHandler
4. Make APPLICATION_POKEMON_STATS constant

@nahojjjen feel free to cherry pick

Blaisorblade commented on app/src/main/java/com/kamron/pogoiv/logic/PokemonShareHandler.java:
```diff
@@ -30,19 +33,19 @@ public void spreadResultIntent(Pokefly pokefly, IVScanResult ivScan) {
 
         } catch (JSONException e) {
-            e.printStackTrace();
+            Log.e(TAG, "Tried to generate jsonPokemon when clicking the share button",e);
```
Please use Timber for logging instead, so we get exception logged on Fabric.

NightMadness commented on app/src/main/java/com/kamron/pogoiv/OcrHelper.java:
```diff
@@ -486,14 +486,14 @@ private String getCandyNameFromImg(Bitmap pokemonImage) {
         String uniqueText = tesseract.getUTF8Text();
 
-        return uniqueText;
+        return Optional.fromNullable(uniqueText);
```
You are right we get empty sting for empty ocr. Optional is not better but its consistant with the rest of the code. Either way is fine with me.

NightMadness replied:
Sounds resonable, no need for it to be optional

nahojjjen commented:
1. Switch string to Optional<String>

Blaisorblade seems to be discussing this change with you

2.Removed pokemonUniqueID from IVScanResult

Sounds good, I was trying to have the uniqueID follow the same standards as hp and cp and other scanned info, but I suppose this also makes sense.

3.Added log to PokemonShareHandler

Good idea, though as blaisorblade said we probably want it to log to fabric, so we get online crashreports.

4.Make APPLICATION_POKEMON_STATS constant

Nice

NightMadness commented:
> 1. Switch string to Optional

> Blaisorblade seems to be discussing this change with you

Removed from this PR.

> 2.Removed pokemonUniqueID from IVScanResult

> Sounds good, I was trying to have the uniqueID follow the same standards as hp and cp and other scanned info, but I suppose this also makes sense.

I think this IVScanResult is mostly useful if we want to use an item with the clipboard or compare previous Scan. but it Its really up to you.

> 3.Added log to PokemonShareHandler
> 
> Good idea, though as blaisorblade said we probably want it to log to fabric, so we get online crashreports.

changed to fabric

> 
> 4.Make APPLICATION_POKEMON_STATS constant
> 
> Nice

Blaisorblade commented:
Re pokemonUniqueId I just realized: do we use it to check when it makes sense to allow "Refine" on scan results? From a superficial look, it seems we don't but maybe we should. Would that be easier if we keep `pokemonUniqueId` in `IVScanResult`?

But otherwise I agree the ID doesn't belong there too much...

nahojjjen commented:
We cant use pokemonUniqueId in the situation where the pokemon is evolved, since that can change type, height and weight.
